clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -264,6 +264,15 @@ public TopicMetadata(Errors error,
             this(error, topic, isInternal, partitionMetadata, 0);
         }
 
+        private static final List<PartitionMetadata> EMPTY_PARTITION_METADATA = new ArrayList<>();
```

Comment:
Thanks for the comment, @Lincong 
1. Yeah, I've made an update to switch to `Collections.emptyList()`
2. About describeTopics(), currently the API does expose partitions info via the `DescribeTopicsResult`, hence changing that behavior could break other services using this API. So instead of changing the current describeTopics() API, I think the best way is to add a new API for the topic existence check.